src/snowflake/cli/api/project/definition_conversion.py

```diff
@@ -144,6 +159,76 @@ def convert_streamlit_to_v2_data(streamlit: Streamlit):
     return data
 
 
+def convert_native_app_to_v2_data(
+    project_root, native_app: NativeApp
+) -> Dict[str, Any]:
+    def _make_meta(obj: Application | Package):
+        meta = {}
+        if obj.role:
+            meta["role"] = obj.role
+        if obj.warehouse:
+            meta["warehouse"] = obj.warehouse
+        if obj.post_deploy:
+            meta["post_deploy"] = obj.post_deploy
+        return meta
+
+    def _find_manifest():
+        # We don't know which file in the project directory is the actual manifest,
+        # and we can't iterate through the artifacts property since the src can contain
+        # glob patterns. The simplest solution is to bundle the app and find the
+        # manifest file from the resultant BundleMap, since the bundle process ensures
+        # that only a single source path can map to the corresponding destination path
+        bundle_map = build_bundle(
+            project_root, Path(native_app.deploy_root), native_app.artifacts
+        )
+        # Use a POSIX path to be consistent with other migrated fields
+        # which use POSIX paths as default values
+        return bundle_map.to_project_path(Path("manifest.yml")).as_posix()
+
+    package_entity_name = "pkg"
+    package = {
+        "type": "application package",
+        "identifier": (
+            native_app.package.name
```

Comment:
Should we suffix the identifier values with `_<% ctx.env.USER %>` to keep the v1 behavior?
@sfc-gh-cgorrie WDYT?

Reply:
We should discuss this in person I think.

Reply:
discussed locally, will followup